src/plugins/kibana_usage_collection/server/collectors/application_usage/fetch_all_saved_objects.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  ISavedObjectsRepository,
+  SavedObjectsCreatePointInTimeFinderOptions,
+  SavedObjectsFindResult,
+} from '@kbn/core/server';
+
+export async function fetchAllSavedObjects<T>(
+  soRepository: ISavedObjectsRepository,
+  findOptions: SavedObjectsCreatePointInTimeFinderOptions
+): Promise<Array<SavedObjectsFindResult<T>>> {
+  const finder = soRepository.createPointInTimeFinder<T>({ ...findOptions, perPage: 100 });
```

Comment:
Overall using a PIT search is definitely an improvement.

I still wonder if `100` is the correct batch size? I would have gone with 1k, to avoid having that much additional roundtrips with ES in case of a lot of SOs

Reply:
I used `100` only because that's the value we [placed in the docs](https://github.com/elastic/kibana/blob/d904aa428d29f3a554466bc6b5397924ae58fdc8/src/core/server/saved_objects/service/saved_objects_client.ts#L676) (so I assumed the recommendation). Happy to change it to any other value.

@lukeelmers as the writer of those docs. What do you think?

Reply:
I can see in the code we are [defaulting to `1000`](https://github.com/elastic/kibana/blob/4d3cab8983df8acbf646fab795f71da1df91704a/src/core/server/saved_objects/service/lib/point_in_time_finder.ts#L89-L93). I'll change it to 1000 then 😇

Reply:
> @lukeelmers as the writer of those docs. What do you think?

Yeah I think I probably just forgot to update the docs when we decided to go with `1000` as the default. There was nothing special about the number `100`.